src/Idris/Package.idr

```diff
@@ -536,30 +536,36 @@ processPackage opts (cmd, file)
                | Left err => throw (FileErr fp err)
              pure ()
         _ =>
-         do let True = isSuffixOf ".ipkg" file
-                 | _ => do coreLift $ putStrLn ("Packages must have an '.ipkg' extension: " ++ show file ++ ".")
-                           coreLift (exitWith (ExitFailure 1))
-            Right (pname, fs) <- coreLift $ parseFile file (parsePkgDesc file <* eoi)
-              | Left err => throw err
-            pkg <- addFields fs (initPkgDesc pname)
-            whenJust (builddir pkg) setBuildDir
-            setOutputDir (outputdir pkg)
-            case cmd of
-              Build => do [] <- build pkg opts
-                            | errs => coreLift (exitWith (ExitFailure 1))
-                          pure ()
-              Install => do [] <- build pkg opts
-                               | errs => coreLift (exitWith (ExitFailure 1))
-                            install pkg opts
-              Typecheck => do [] <- check pkg opts
-                                | errs => coreLift (exitWith (ExitFailure 1))
+          do Just (dir, filename) <- pure $ splitParent file
+               | Nothing => throw $ InternalError "Tried to split empty string"
+             True <- pure $ isSuffixOf ".ipkg" filename
+               | False => coreLift . putStrLn $ "Packages must have an '.ipkg' extension: " ++ show file ++ "."
+             setWorkingDir dir
+             Right (pname, fs) <- coreLift $ parseFile filename
+                                      (do desc <- parsePkgDesc filename
+                                          eoi
+                                          pure desc)
+                | Left err => throw err
+             pkg <- addFields fs (initPkgDesc pname)
+             maybe (pure ()) setBuildDir (builddir pkg)
+             setOutputDir (outputdir pkg)
+             case cmd of
+                  Build => do [] <- build pkg opts
+                                 | errs => coreLift (exitWith (ExitFailure 1))
                               pure ()
-              Clean => clean pkg opts
-              REPL => do [] <- build pkg opts
-                            | errs => coreLift (exitWith (ExitFailure 1))
-                         runRepl (map snd $ mainmod pkg)
-              Init => pure () -- already handled earlier
-
+                  Install => do [] <- build pkg opts
+                                   | errs => coreLift (exitWith (ExitFailure 1))
+                                install pkg opts
+                  Typecheck => do
+                    [] <- check pkg opts
+                      | errs => coreLift (exitWith (ExitFailure 1))
+                    pure ()
+                  Clean => clean pkg opts
+                  REPL => do
+                    [] <- build pkg opts
+                       | errs => coreLift (exitWith (ExitFailure 1))
+                    runRepl (map snd $ mainmod pkg)
```

Comment:
Given that in this case we open a repl, does it make sense to require that there is only
one package file passed as an argument?

Reply:
Not really, but that gets complicated. `idris2 --build foo.ipkg bar.ipkg` also makes sense to me.

However, `idris2 --repl foo.ipkg --repl bar.ipkg` acts exactly the same as `idris2 --repl foo.ipkg`, which probably not the right behaviour.

Should certain options, like `--repl`, `--build`, `-p`, etc; take `n>=1` arguments?